test/test_nn.py

```diff
@@ -11351,6 +11390,116 @@ def test_maxpool3d_non_square_backward(self, device):
             self.assertTrue(torch.allclose(x.grad, torch.ones_like(x.grad)))
 
 
+class TestConv2dExt(TestCase):
+    def test_conv2d_ext(self):
+        def subtensor(tensor, dim, groups, g):
```

Comment:
Can you extract these helper functions so the test doesn't include them? That would help clarify what's being tested. You can put them in the class with a _ before their names.